src/test/scala/PropertyCheckerTest.scala

```diff
@@ -631,6 +631,137 @@ class PropertyCheckerTest extends FunSuite {
   // are implemented. Currently the exceptions raised when these properties are missing is not tested since
   // NoSuchElementExceptio is thrown when a jsonnode with these properties are passed in.
 
+  test("set invalid value warnings when Uri template property is not string") {
+    val json = """
+                 |{
+                 |  "sampleObjectProperty": "some content"
+                 | }
+                 |""".stripMargin
+    val jsonNode = objectMapper.readTree(json)
+    val (_, warnings, _) = PropertyChecker.checkProperty("propertyUrl", jsonNode, baseUrl = "https://chickenburgers.com", "und")
+
+    assert(warnings === Array[String]("invalid_value"))
+  }
+
+  test("set value and warnings correctly when Uri template property valid") {
+    val validTextNodeUrl = new TextNode("https://www.w3.org")
+    val (values, warnings, _) = PropertyChecker.checkProperty("propertyUrl", validTextNodeUrl , baseUrl = "https://chickenburgers.com", "und")
+
+    assert(values === validTextNodeUrl)
+    assert(warnings === Array[String]())
+  }
+
+  test("set correct value and warnings correctly when textDirection property is valid") {
+    val validTextDirection = new TextNode("rtl")
+    val (values, warnings, _) = PropertyChecker.checkProperty("textDirection", validTextDirection , baseUrl = "https://www.w3.org", "und")
+
+    assert(values === validTextDirection)
+    assert(warnings === Array[String]())
+  }
+
+  test("set correct value and warnings correctly when textDirection property is Invalid") {
+    val validTextDirection = new TextNode("Some value which is not a text direction")
+    val (values, warnings, _) = PropertyChecker.checkProperty("textDirection", validTextDirection , baseUrl = "https://www.w3.org", "und")
+
+    assert(warnings === Array[String]("invalid_value"))
+  }
+
+  // Title Property tests
+  test("set lang object when value is textual in title property") {
+    val (values, warnings, _) = PropertyChecker.checkProperty("titles", new TextNode("Sample Title"), "", "und")
+
+    assert(values.path("lang").isMissingNode === false)
```

Comment:
I hate when people do `== true` and `== false`. Please don't. It makes it look like we're amateurs IMO.

Reply:
Okay, changed such assertions